flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/JsonFunctionsITCase.java

```diff
@@ -420,8 +431,146 @@ private static TestSpec jsonValue() throws Exception {
                                 "No results for path"));
     }
 
+    private static List<TestSpec> jsonObjectSpec() {
+        final Map<String, String> mapData = new HashMap<>();
+        mapData.put("M1", "V1");
+        mapData.put("M2", "V2");
+
+        final Map<String, Integer> multisetData = new HashMap<>();
+        multisetData.put("M1", 1);
+        multisetData.put("M2", 2);
+
+        return Arrays.asList(
+                TestSpec.forFunction(BuiltInFunctionDefinitions.JSON_OBJECT)
+                        .onFieldsWithData(0)
+                        .testResult(
+                                jsonObject(JsonOnNull.NULL),
+                                "JSON_OBJECT()",
+                                "{}",
+                                STRING().notNull(),
+                                VARCHAR(2000).notNull())
+                        .testResult(
+                                jsonObject(JsonOnNull.NULL, "K", nullOf(STRING())),
+                                "JSON_OBJECT(KEY 'K' VALUE CAST(NULL AS STRING) NULL ON NULL)",
+                                "{\"K\":null}",
+                                STRING().notNull(),
+                                VARCHAR(2000).notNull())
+                        .testResult(
+                                jsonObject(JsonOnNull.ABSENT, "K", nullOf(STRING())),
+                                "JSON_OBJECT(KEY 'K' VALUE CAST(NULL AS STRING) ABSENT ON NULL)",
+                                "{}",
+                                STRING().notNull(),
+                                VARCHAR(2000).notNull()),
+                TestSpec.forFunction(BuiltInFunctionDefinitions.JSON_OBJECT)
+                        .onFieldsWithData(
+                                "V",
+                                true,
+                                1,
+                                1.23d,
+                                1.23,
+                                LocalDateTime.parse("1990-06-02T13:37:42.001"),
+                                Instant.parse("1990-06-02T13:37:42.001Z"),
+                                Arrays.asList("A1", "A2", "A3"),
+                                Row.of("R1"),
```

Comment:
we should add at least a second field, maybe with a timestamp?

Reply:
Done. I also added another example which nestes several complex types as that revealed another minor bug in the implementation.